src/core/thread/thread_netif.cpp

```diff
@@ -94,18 +94,16 @@ const char *ThreadNetif::GetName(void) const
 
 ThreadError ThreadNetif::Up(void)
 {
-    ThreadError error = kThreadError_None;
-
-    VerifyOrExit(!mIsUp, error = kThreadError_Already);
-
-    mIp6.AddNetif(*this);
-    mMeshForwarder.Start();
-    mCoapServer.Start();
-    mMleRouter.Enable();
-    mIsUp = true;
+    if (!mIsUp)
```

Comment:
why not just change
`VerifyOrExit(!mIsUp, error = kThreadError_Already);`
to
`VerifyOrExit(!mIsUp, error = kThreadError_None);`